Add RPC failover options (#2140)
near/docs: docs/5.api/rpc/providers.md

| -------------------------------------------------------------------------- | ------------------------------------------------------------ |
| [NEAR](setup.md)                                                           | `https://rpc.testnet.near.org`                               |
| [Pagoda](https://www.pagoda.co/console)                                    | `https://rpc.testnet.pagoda.co`                              |

## RPC Failover

> In `near-api-js` you can use [FailoverRpcProvider](../../tools/near-api-js/quick-reference#rpc-failover) to automatically switch RPC providers when one provider is experiencing downtime, or implement an RPC selection widget that allows users to add their own RPC provider.

Reviewer: ```suggestion
In `near-api-js` you can use [`FailoverRpcProvider`](../../4.tools/near-api-js/quick-reference.md#rpc-failover) to automatically switch RPC providers when one provider is experiencing downtime, or implement an RPC selection widget that allows users to add their own RPC provider.
```

> As a user, if some dapp or wallet doesn't support RPC failover and the primary provider is down, you can use [RPC Selector](https://github.com/Sliman4/rpc-selector) browser extension to redirect all requests from providers in the list above to the RPC provider of your choice.

Reviewer: ```suggestion
As a user, if a dApp or wallet doesn't support RPC failover and the primary provider is down, you can use an RPC Selector browser extension to redirect all requests to an RPC provider of your choice.
```

Author: Do you think it's worth mentioning the extension if there's no link? This extension with 3 users is surprisingly well googleable using "rpc selector" term, but I'm not sure if anyone is going to even try to find it. Personally, I never try googling anything with such a generic name. The whole reason I made this PR is that people in the dev community found it useful ([1](https://t.me/neardev/51431), [2](https://t.me/neardev/51432)), and removing the link defeats the entire purpose. If the issue is that it links to a third-party-owned repo, I can transfer ownership, I'm not actively maintaining it anyway

Reviewer: We can't reference such a tool without a proper security audit, as this could impact user's transactions. (Even after a repository transfer, an audit would be required)
